runtime/lua/vim/treesitter/languagetree.lua

```diff
@@ -15,6 +15,15 @@ LanguageTree.__index = LanguageTree
 ---@param opts.injections A table of language to injection query strings.
 ---                      This is useful for overriding the built-in runtime file
 ---                      searching for the injection language query per language.
+---@param opts.injection_resolver(lang, context)
```

Comment:
probably this doesn't need to be per language tree but could also be a module level function (publicly visible).